File: common/src/main/java/net/msimod/common/networking/socket/Message.java
package net.msimod.common.networking.socket;

import java.io.Serializable;

/**
 * A message that can be sent over the socket.
 */
public class Message implements Serializable {
    /**
     * The handlers for the socket
     */
    public static enum Handlers {
        NONE,
        AUTH,
        CHAT;
    }

    /**
     * The type of the message
     */
    public static enum Type {
        REGISTER,
        UPDATE,
        UNREGISTER,
        FAILURE,
        UNAUTHORIZED,
    }

    public Message(Handlers handler, Type type, Object data) {
        this.handler = handler;
        this.type = type;
        this.data = data;
    }

    public Handlers handler;
    public Type type;
    public String accessToken;
    public Object data;
}

File: webclient/src/lib/socket/message.js
/**
 * Message hadlers
 */
export const MessageHandlers = {
    NONE: 'NONE',
    AUTH: 'AUTH',
    CHAT: 'CHAT',
};

/**
 * Message types
 */
export const MessageTypes = {
    REGISTER: 'REGISTER',
    UPDATE: 'UPDATE',
    UNREGISTER: 'UNREGISTER',
    FAILURE: 'FAILURE',
    UNAUTHORIZED: 'UNAUTHORIZED',
};

/**
 * @class Message
 * @description A message sent from the client to the server
 * @property {string} handler The handler for the message
 * @property {string} type The type of the message
 * @property {string?} accessToken The access token of the message
 * @property {any} data The data of the message
 */
export class Message {
    constructor(handler, type, accessToken, data) {
        this.handler = handler;
        this.type = type;
        this.accessToken = accessToken;
        this.data = data;
    }
}